elephant/online.py

```diff
@@ -0,0 +1,770 @@
+                coincidences by spike time randomization in each trial and
+                sum over trials
+        'analytic_TrialAverage' and 'surrogate_TrialByTrial' are not supported
+        yet.
+        Default: 'analytic_trialByTrial'
+    n_surrogates : int
+        Number of surrogates which would be used when 'surrogate_TrialByTrial'
+        is chosen as 'method'. Yet 'surrogate_TrialByTrial' is not supported.
+    input_parameters : dict
+        Dictionary of the input parameters which would be used for calling
+        the offline version of UEA for the same data to get the same results.
+    Js : np.ndarray
+        JointSurprise of different given patterns within each window.
+            * 0-axis --> different window
+            * 1-axis --> different pattern hash
+    n_exp : np.ndarray
+        The expected number of coincidences of each pattern within each window.
+            * 0-axis --> different window
+            * 1-axis --> different pattern hash
+    n_emp : np.ndarray
+        The empirical number of coincidences of each pattern within each
+        window.
+            * 0-axis --> different window
+            * 1-axis --> different pattern hash
+    rate_avg : np.ndarray
+        The average firing rate of each neuron of each pattern within
+        each window.
+            * 0-axis --> different window
+            * 1-axis --> different pattern hash
+            * 2-axis --> different neuron
+    indices : defaultdict
+        Dictionary contains for each trial the indices of pattern
+        within each window.
+
+    Methods
+    -------
+    get_results()
+        Returns the result dictionary with the following class attribute names
+        as keys and the corresponding attribute values as the complementary
+        value for the key: (see also Attributes section for respective key
+        descriptions)
+        * 'Js'
+        * 'indices'
+        * 'n_emp'
+        * 'n_exp'
+        * 'rate_avg'
+        * 'input_parameters'
+    update_uea(spiketrains, events)
+        Updates the entries of the result dictionary by processing the
+        new arriving 'spiketrains' and trial defining trigger 'events'.
+    reset(bw_size, trigger_events, trigger_pre_size, trigger_post_size,
+            saw_size, saw_step, n_neurons, pattern_hash, time_unit,
+            save_n_trials)
+        Resets all class attributes to their initial (default) value. It is
+        actually a re-initialization which allows parameter adjustments.
+    get_all_saved_trials()
+        Returns the last 'n'-trials which were analyzed according to the FIFO
+        strategy (first in, first out).
+
+    Returns
+    -------
+    see 'get_results()' in Methods section
+
+    Notes
+    -----
+    Common abbreviations which are used in both code and documentation:
+        bw = bin window
+        tw = trial window
+        saw = sliding analysis window
+        idw = incoming data window
+        mw = memory window
```

Comment:
in some cases abbreviations are used in elephant, if you want to keep them, please help to justify their use in this case.